[Bug]: Client Crash when building Redstone Data Interface in Multiplayer

### Initial Check

- [x] I have checked the existing issues and this one is not a duplicate

### Category

🧱Multiblocks

### Side

Client

### Description

Whenever forming the Multiblock of the Redstone Data Interface in Multiplayer, the Client crashes.
When rejoining the Server, the Multiblock has been formed and works.

Relevant Part of the Crashlog:
```
...
[16:26:31] [Client thread/FATAL] [minecraft/Minecraft]: Reported exception thrown!
net.minecraft.util.ReportedException: Ticking block entity
        at net.minecraft.world.World.updateEntities(World.java:2026) ~[World.class:?]
        at net.minecraft.client.Minecraft.runTick(Minecraft.java:1974) ~[Minecraft.class:?]
        at net.minecraft.client.Minecraft.runGameLoop(Minecraft.java:1188) ~[Minecraft.class:?]
        at net.minecraft.client.Minecraft.run(Minecraft.java:442) [Minecraft.class:?]
        at net.minecraft.client.main.Main.main(Main.java:118) [Main.class:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_201]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_201]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_201]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_201]
        at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
        at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_201]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_201]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_201]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_201]
        at net.minecraftforge.gradle.GradleStartCommon.launch(GradleStartCommon.java:98) [start/:?]
        at GradleStart.main(GradleStart.java:26) [start/:?]
Caused by: java.lang.StackOverflowError
        at pl.pabilo8.immersiveintelligence.common.util.multiblock.TileEntityMultiblockIIBase.getUUID(TileEntityMultiblockIIBase.java:506) ~[TileEntityMultiblockIIBase.class:?]
        at pl.pabilo8.immersiveintelligence.common.util.multiblock.TileEntityMultiblockIIBase.getUUID(TileEntityMultiblockIIBase.java:506) ~[TileEntityMultiblockIIBase.class:?]
        at pl.pabilo8.immersiveintelligence.common.util.multiblock.TileEntityMultiblockIIBase.getUUID(TileEntityMultiblockIIBase.java:506) ~[TileEntityMultiblockIIBase.class:?]
...

```

### Reproduction

* Place all the needed Blocks for the Redstone Data Interface
* Use the Hammer to complete the Multiblock
* _Witness the Crash_
* Restart Minecraft
* Rejoin Server
* Multiblock is there and works

### Expected Behaviour

The Game shouldn't crash when forming the Multiblock

### Logs

_No response_

### Operating System

Linux

### Minecraft Version

1.12.2

### Immersive Intelligence Version

0.3.1-dev3-test1

### Immersive Engineering Version

0.98

--- a/src/main/java/pl/pabilo8/immersiveintelligence/common/util/multiblock/TileEntityMultiblockIIBase.java
+++ b/src/main/java/pl/pabilo8/immersiveintelligence/common/util/multiblock/TileEntityMultiblockIIBase.java
@@ -500,27 +500,12 @@ public abstract class TileEntityMultiblockIIBase<T extends TileEntityMultiblockI
 
 	public UUID getUUID()
 	{
-		if(isDummy())
+		if(this.uuid==null)
 		{
-			T master = master();
-			if (master == null)
-			{
-				return IIUtils.getBlockPosUUID(getPos());
-			} else
-			{
-				if (world.isRemote)
-				{
-					return master.uuid;
-				} else
-				{
-					return master.getUUID();
-				}
-
-			}
-		}
-		if (this.uuid == null)
-		{
-			this.uuid = IIUtils.getBlockPosUUID(getPos());
+			UUID uuid = IIUtils.getBlockPosUUID(getPos().add(-offset[0], -offset[1], -offset[2]));
+			if(master!=null)
+				this.uuid = uuid; //Master exists, so the position is correct
+			return uuid;
 		}
 		return this.uuid;
 	}

--- a/src/main/java/pl/pabilo8/immersiveintelligence/common/util/multiblock/TileEntityMultiblockIIBase.java
+++ b/src/main/java/pl/pabilo8/immersiveintelligence/common/util/multiblock/TileEntityMultiblockIIBase.java
@@ -503,9 +503,16 @@ public abstract class TileEntityMultiblockIIBase<T extends TileEntityMultiblockI
 		if(isDummy())
 		{
 			T master = master();
-			return master==null?IIUtils.getBlockPosUUID(getPos()): master.getUUID();
+			if (master == null) {
+				return IIUtils.getBlockPosUUID(getPos());
+			} else {
+				return master.uuid;
+			}
 		}
-		return this.uuid==null?this.uuid = IIUtils.getBlockPosUUID(getPos()): this.uuid;
+		if (this.uuid == null) {
+			this.uuid = IIUtils.getBlockPosUUID(getPos());
+		}
+		return this.uuid;
 	}
 
 	public long getTicksExisted()

--- a/src/main/java/pl/pabilo8/immersiveintelligence/common/util/multiblock/TileEntityMultiblockIIBase.java
+++ b/src/main/java/pl/pabilo8/immersiveintelligence/common/util/multiblock/TileEntityMultiblockIIBase.java
@@ -506,7 +506,12 @@ public abstract class TileEntityMultiblockIIBase<T extends TileEntityMultiblockI
 			if (master == null) {
 				return IIUtils.getBlockPosUUID(getPos());
 			} else {
-				return master.uuid;
+				if (world.isRemote) {
+					return master.uuid;
+				} else {
+					return master.getUUID();
+				}
+
 			}
 		}
 		if (this.uuid == null) {

--- a/src/main/java/pl/pabilo8/immersiveintelligence/common/util/multiblock/TileEntityMultiblockIIBase.java
+++ b/src/main/java/pl/pabilo8/immersiveintelligence/common/util/multiblock/TileEntityMultiblockIIBase.java
@@ -503,18 +503,23 @@ public abstract class TileEntityMultiblockIIBase<T extends TileEntityMultiblockI
 		if(isDummy())
 		{
 			T master = master();
-			if (master == null) {
+			if (master == null)
+			{
 				return IIUtils.getBlockPosUUID(getPos());
-			} else {
-				if (world.isRemote) {
+			} else
+			{
+				if (world.isRemote)
+				{
 					return master.uuid;
-				} else {
+				} else
+				{
 					return master.getUUID();
 				}
 
 			}
 		}
-		if (this.uuid == null) {
+		if (this.uuid == null)
+		{
 			this.uuid = IIUtils.getBlockPosUUID(getPos());
 		}
 		return this.uuid;

